README.md

```diff
@@ -1,4 +1,182 @@
+
+# Installation
+
+To install fully functional RADAR-base data uploader from source, please use the `docker-compose.yml` under the root directory
+
+```bash
+docker-compose up -d --build
+```
+
+
+# Usage
+
+## Configuring RADAR-base Data Uploader
+
+**Disclaimer:** This section does not cover how to install or configure the complete RADAR-Base stack. Please visit [RADAR-Docker](https://github.com/RADAR-base/RADAR-Docker) or [RADAR-Kubernetes](https://github.com/RADAR-base/RADAR-Kubernetes) for complete installation guidelines. 
+
+**Note:** Some configuration values are specified under the assumption that the docker containers will have the same name on deployment as well.
+
+Configuring Data Uploader involves configuring the three components mentioned above.
+
+### 1. Configuring radar-upload-backend
+Please copy `etc/upload.yml.template` to `etc/upload.yml` and modify the database credentials and the oauth client credentials. The following tables shows the possible properties and explanation.
+
+```yaml
+
+baseUri: "http://0.0.0.0:8085/upload/api/"
+managementPortalUrl: "http://managementportal-app:8080/managementportal/"
+jdbcDriver: "org.postgresql.Driver"
+jdbcUrl: "jdbc:postgresql://radarbase-postgresql:5432/uploadconnector"
+jdbcUser: "username" # change this to your database username
+jdbcPassword: "password" # change this to your database password
+additionalPersistenceConfig:
+  "hibernate.dialect": "org.hibernate.dialect.PostgreSQL95Dialect"
+enableCors: yes # if you want to enable cors filter to this component specify yes
+clientId: "radar_upload_backend" # configure your oauth client id
+clientSecret: "secret" # configure the client-secret
+sourceTypes: # these are the data source types that are supported to upload data
+  - name: "phone-acceleration"
+    topics:
+      - "android_phone_acceleration"
+    contentTypes:
+      - "text/csv"
+    timeRequired: false
+    sourceIdRequired: false
+    configuration:
+      "setting1": "value1"
+      "setting2": "value2"
+  - name: "acceleration-zip"
+    topics:
+      - "android_phone_acceleration"
+    contentTypes:
+      - "application/zip"
+    timeRequired: false
+    sourceIdRequired: false
+  - name: "altoida-zip"
+    topics:
+      - "connect_upload_altoida_acceleration"
+    contentTypes:
+      - "application/zip"
+    timeRequired: false
+    sourceIdRequired: false
+
+```
+
+#### Adding support to new device type
+To add support to additional device types, add a new entry of sourceType to the `sourceTypes` list.
+A single sourceType entry is defined as below.
+
+```yaml
+
+- name: "acceleration-zip" # unique identifier of the data source or device (*required)
+    topics: # list of topics to send data
+      - "android_phone_acceleration" 
+    contentTypes: # content types of the data
+      - "application/zip"
+    timeRequired: false # if the data is large or if uploading data would require a lot of time mention this to true.
```

Comment:
This field was meant to indicate whether the user should supply a date-time for the upload, that is, that the uploaded content does not contain the relevant timestamp.

Reply:
Oh I see. That makes sense. I will change it